freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/DetailedStackStatus.java

```diff
@@ -83,7 +83,11 @@ public enum DetailedStackStatus {
     // CCM upgrade statuses
     UPGRADE_CCM_REQUESTED(Status.UPGRADE_CCM_REQUESTED, AvailabilityStatus.AVAILABLE),
     UPGRADE_CCM_IN_PROGRESS(Status.UPGRADE_CCM_IN_PROGRESS, AvailabilityStatus.AVAILABLE),
-    UPGRADE_CCM_FAILED(Status.UPGRADE_CCM_FAILED, AvailabilityStatus.UNAVAILABLE);
+    UPGRADE_CCM_FAILED(Status.UPGRADE_CCM_FAILED, AvailabilityStatus.UNAVAILABLE),
+    // Modify proxy config statuses
+    MODIFY_PROXY_CONFIG_REQUESTED(Status.MODIFY_PROXY_CONFIG_REQUESTED, AvailabilityStatus.AVAILABLE),
+    MODIFY_PROXY_CONFIG_IN_PROGRESS(Status.MODIFY_PROXY_CONFIG_IN_PROGRESS, AvailabilityStatus.AVAILABLE),
+    MODIFY_PROXY_CONFIG_FAILED(Status.MODIFY_PROXY_CONFIG_FAILED, AvailabilityStatus.UNAVAILABLE);
```

Comment:
why will be the freeipa unavailable if it doesn't effect any existing instances?

Reply:
this will be useful when the existing instances will be modified too, but I have added it now with copying CCM steps/statuses/etc.

Reply:
maybe this should be done by migrating one instance at a time so you could revert the change and also maintain connectivity during the process. So I think this shouldn't be unavailable.